Solutions implemented:

NikolayAbysov opened the pull request:
1 Removing product from shopping cart
2 Order button in shopping cart
3 List of orders
4 Order deletion
5 Two order pages (for customer and admin)
6 Bootstrap is on the half way :)

NikolayAbysov committed:
Solutions implemented:
1 Removing product from shopping cart
2 Order button in shopping cart
3 List of orders
4 Order deletion
5 Two order pages (for customer and admin)
6 Bootstrap is on the half way :)

Tarabarnik commented on src/main/java/mate/academy/internetshop/controllers/DeleteProductFromShoppingCartController.java:
```diff
@@ -0,0 +1,30 @@
+package mate.academy.internetshop.controllers;
```
```suggestion
package mate.academy.internetshop.controller;
```
Пекеджі потрібно називати в однині

NikolayAbysov replied:
Done

Tarabarnik commented on src/main/java/mate/academy/internetshop/controllers/DeleteProductFromShoppingCartController.java:
```diff
@@ -0,0 +1,30 @@
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        ShoppingCart shoppingCart
+                = shoppingCartService.getByUserId(Long.valueOf(req.getParameter("userId")));
```
поки що, використовуємо заглушку USER_ID

NikolayAbysov replied:
Done

Tarabarnik commented on src/main/java/mate/academy/internetshop/controllers/OrderController.java:
```diff
@@ -0,0 +1,54 @@
+import mate.academy.internetshop.service.ShoppingCartService;
+import mate.academy.internetshop.service.UserService;
+
+public class OrderController extends HttpServlet {
```
```suggestion
public class CompleteOrderController extends HttpServlet {
```

Tarabarnik commented on src/main/java/mate/academy/internetshop/controllers/OrderController.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        User user = userService.get(USER_ID);
```
надто складна реалізація. отримай корзину по заглушці USER_ID, потім виклич метод completeOrder і передай туди товари та юзера із корзини. а редіректити треба на сторінку, де буде інфо лише по новому ордеру

Tarabarnik replied:
інфо по окремому замовленню теж потрібно винести в окремий контролер

Tarabarnik commented on src/main/java/mate/academy/internetshop/controllers/ProductsController.java:
```diff
@@ -20,6 +20,15 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         req.setAttribute("products", products);
-        req.getRequestDispatcher("/WEB-INF/views/products/all.jsp").forward(req, resp);
+
+        if (req.getParameter("get") != null && req.getParameter("get").equals("all")) {
```
для видалення треба створити окремий контролер, по single responsibility principle

NikolayAbysov committed:
Issues fixed:

1 Logic from one OrderController divided to 4 controllers: viewOrder, viewAllOrders, deleteOrder and CompleteOrder

Tarabarnik commented on src/main/java/mate/academy/internetshop/controller/CompleteOrderController.java:
```diff
@@ -0,0 +1,36 @@
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        User user = userService.get(USER_ID);
+        ShoppingCart shoppingCart
+                = shoppingCartService.getByUserId(USER_ID);
+        Order order = orderService.completeOrder(shoppingCart.getProducts(), user);
```
```suggestion
        ShoppingCart shoppingCart
                = shoppingCartService.getByUserId(USER_ID);
        Order order = orderService.completeOrder(shoppingCart.getProducts(), shoppingCart.getUser());
```

NikolayAbysov committed:
Update src/main/java/mate/academy/internetshop/controller/CompleteOrderController.java

Co-authored-by: Pavel Kurilyuk <50627236+Tarabarnik@users.noreply.github.com>
